graph/appender/dead_node.go

```diff
@@ -12,76 +13,64 @@ import (
 //   edges (kiali-1326). Kiali-1526 adds an exclusion to this rule. Egress is a special
 //   terminal service node and we want to show it even if it has no incoming traffic
 //   for the time period.
-type DeadNodeAppender struct {
-	// externalServices acts as a cache. This prevents from querying multiple times
-	// the cluster for service entries when the graph is requested for all namespaces.
-	externalServices map[string]bool
-}
+type DeadNodeAppender struct{}
 
 // AppendGraph implements Appender
-func (a DeadNodeAppender) AppendGraph(trafficMap graph.TrafficMap, _ string) {
+func (a DeadNodeAppender) AppendGraph(trafficMap graph.TrafficMap, globalInfo *GlobalInfo, namespaceInfo *NamespaceInfo) {
 	if len(trafficMap) == 0 {
 		return
 	}
 
-	business, err := business.Get()
-	checkError(err)
-
-	if a.externalServices == nil {
-		a.externalServices = resolveExternalServices(business)
+	var err error
+	if globalInfo.IstioClient == nil {
+		globalInfo.IstioClient, err = kubernetes.NewClient()
```

Comment:
If an appender needs to access to kubernetes package, then it seems we need to move something to the business layer.